pvlib/spectrum/mismatch.py

```diff
@@ -0,0 +1,213 @@
+                        [ 650, 0.71],
+                        [ 800, 0.88],
+                        [ 900, 0.97],
+                        [ 950, 1.00],
+                        [1000, 0.93],
+                        [1050, 0.58],
+                        [1100, 0.21],
+                        [1150, 0.05],
+                        [1190, 0.00]]).transpose()
+
+    if wavelength is None:
+        resolution = 5.0
+        wavelength = np.arange(280, 1200 + resolution, resolution)
+
+    interpolator = interp1d(SR_DATA[0], SR_DATA[1],
+                            kind='cubic',
+                            bounds_error=False,
+                            fill_value=0.0,
+                            copy=False,
+                            assume_sorted=True)
+
+    sr = pd.Series(data=interpolator(wavelength), index=wavelength)
+
+    sr.index.name = 'wavelength'
+    sr.name = 'sr'
+
+    return sr
+
+
+def get_am15g(wavelength=None):
+    '''
+    Read the ASTM G173-03 AM1.5 global tilted spectrum, optionally interpolated
+    to the specified wavelength(s).
+
+    Parameters
+    ----------
+    wavelength: 1-D sequence of numeric, optional
+        The wavelengths at which the spectrum should be interpolated.
+        By default the 2002 wavelengths of the standard are returned. [nm]
+
+    Returns
+    -------
+    am15g: pandas.Series
+        The AM1.5g standard spectrum indexed by wavelength in nm. [W/m^2/nm]
+
+    Notes
+    -----
+    This function uses linear interpolation.  If the reference spectrum is too
+    coarsely interpolated, its integral may deviate from the standard value
+    of 1000.37 W/m^2.
+
+    The values in the data file provided with pvlib-python are copied from an
+    Excel file distributed by NREL, which is found here:
+    https://www.nrel.gov/grid/solar-resource/assets/data/astmg173.xls
+
+    More information about reference spectra is found here:
+    https://www.nrel.gov/grid/solar-resource/spectra-am1.5.html
+
+    References
+    ----------
+    .. [1] ASTM "G173-03 Standard Tables for Reference Solar Spectral
+        Irradiances: Direct Normal and Hemispherical on 37° Tilted Surface."
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Aug. 2022
+
+    pvlib_path = pvlib.__path__[0]
+    filepath = os.path.join(pvlib_path, 'data', 'astm_g173_am15g.csv')
+
+    am15g = pd.read_csv(filepath, index_col=0, squeeze=True)
+
+    if wavelength is not None:
+        interpolator = interp1d(am15g.index, am15g,
+                                kind='linear',
+                                bounds_error=False,
+                                fill_value=0.0,
+                                copy=False,
+                                assume_sorted=True)
+
+        am15g = pd.Series(data=interpolator(wavelength), index=wavelength)
+
+    am15g.index.name = 'wavelength'
+    am15g.name = 'am15g'
+
+    return am15g
+
+
+def calc_spectral_mismatch(sr, e_sun, e_ref=None):
```

Comment:
I think the name of this function should somehow reflect that it is specific to the case of a broadband reference device.

Reply:
How about: ```calc_spectral_mismatch_to_broadband()```

Reply:
Actually, I am going to update my suggestion to: ```calc_field_spectral_mismatch()```

Reply:
I would prefer `calc_spectral_mismatch_field` to make space for `_lab`, someday.

Reply:
Another idea: keep the function general (not specifically relative to broadband reference) add an optional `sr_ref` parameter that defaults to 1.0?

Reply:
Its a very reasonable suggestion, but perhaps it could be done in a future PR with some more debate about the range of interesting options and different ways of implementing them?